mongo/datadog_checks/mongo/mongo.py

```diff
@@ -117,6 +118,15 @@ def __init__(self, name, init_config, instances=None):
             self.auth_source,
         ) = parse_mongo_uri(self.server, sanitize_username=bool(self.ssl_params))
 
+        if self.is_arbiter:
+            # For arbiters, removes the username and password from the connection string.
+            self.server = build_connection_string(
```

Comment:
Here we're re-building the connection string. A better option could be to extract the username and password from the connection string and remove them.